src/api/ruleData/ruleData.service.spec.ts

```diff
@@ -53,10 +65,22 @@ describe('RuleDataService', () => {
     expect(await service.getAllRuleData()).toEqual(result);
   });
 
+  it('should return a rule draft for a given ruleId', async () => {
+    expect(await service.getRuleDataWithDraft(mockRuleData._id)).toEqual(mockRuleDraft);
+  });
+
   it('should get data for a rule', async () => {
     expect(await service.getRuleData(mockRuleData._id)).toEqual(mockRuleData);
   });
 
+  it('should update rule data', async () => {
+    // TODO: Implement
+  });
+
+  it('should delete rule data', async () => {
+    // TODO: Implement
```

Comment:
I don't want to make a regular practice of this, but this was taking me too long and wanted to wrap as much up before end of sprint as possible